RevenueCatUI/Views/FooterView.swift

```diff
@@ -167,8 +167,11 @@ private struct RestorePurchasesButton: View {
 
     var body: some View {
         AsyncButton {
-            _ = try await self.purchaseHandler.restorePurchases()
-            self.displayRestoredAlert = true
+            let customerInfo = try await self.purchaseHandler.restorePurchases()
+
+            if !customerInfo.entitlements.active.isEmpty {
+                self.displayRestoredAlert = true
+            }
```

Comment:
This might not be accurate - it assumes that the developer is using entitlements, which is encouraged but not required. 
We should check nonSubscriptionTransactions as well

Reply:
Good point.